File: examples/python/upload_file.py
# Hippius S3 — Upload files with boto3
# Docs: /storage/s3/python

import os
from dotenv import load_dotenv
import boto3
from botocore.config import Config

load_dotenv()

s3 = boto3.client(
    "s3",
    endpoint_url="https://s3.hippius.com",
    aws_access_key_id=os.environ["HIPPIUS_ACCESS_KEY"],
    aws_secret_access_key=os.environ["HIPPIUS_SECRET_KEY"],
    region_name="decentralized",
    config=Config(
        signature_version="s3v4",
        s3={"addressing_style": "path"},
    ),
)

bucket_name = "my-bucket"

# Upload from disk
s3.upload_file("local-file.txt", bucket_name, "remote-name.txt")
print("Uploaded local-file.txt")

# Upload from memory
s3.put_object(
    Bucket=bucket_name,
    Key="hello.txt",
    Body=b"Hello from Hippius!",
)
print("Uploaded hello.txt from memory")

File: examples/javascript/upload-file.mjs
// Hippius S3 — Upload files with AWS SDK v3
// Docs: /storage/s3/javascript

import "dotenv/config";
import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
import { readFileSync } from "fs";

const s3 = new S3Client({
  endpoint: "https://s3.hippius.com",
  region: "decentralized",
  credentials: {
    accessKeyId: process.env.HIPPIUS_ACCESS_KEY,
    secretAccessKey: process.env.HIPPIUS_SECRET_KEY,
  },
  forcePathStyle: true,
});

const bucketName = "my-bucket";

// Upload from disk
await s3.send(new PutObjectCommand({
  Bucket: bucketName,
  Key: "file.txt",
  Body: readFileSync("./file.txt"),
}));
console.log("Uploaded file.txt from disk");

// Upload from string
await s3.send(new PutObjectCommand({
  Bucket: bucketName,
  Key: "hello.txt",
  Body: "Hello from Hippius!",
  ContentType: "text/plain",
}));
console.log("Uploaded hello.txt from string");
